cmake/projects/OpenSSL/schemes/url_sha1_openssl_macos.cmake.in

```diff
@@ -0,0 +1,220 @@
+    ${ssl_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl${libsuffix}"
+    COMMAND
+    lipo
+    -create
+    ${crypto_libraries}
+    -o
+    "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libcrypto${libsuffix}"
+)
+
+if(BUILD_SHARED_LIBS)
+  # Fix the dylib loading paths
+  ExternalProject_Add(
+      "@HUNTER_EP_NAME@-fix_install_name"
+      DOWNLOAD_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy download command"
+      DOWNLOAD_DIR
+      "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+      TLS_VERIFY
+      "@HUNTER_TLS_VERIFY@"
+      SOURCE_DIR
+      "@HUNTER_PACKAGE_SOURCE_DIR@/macos_universal"
+      INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+          # not used, just avoid creating Install/<name> empty directory
+      CONFIGURE_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy configure command"
+      BUILD_COMMAND
+      "@CMAKE_COMMAND@" -E echo "Dummy build command"
+      INSTALL_COMMAND
+      install_name_tool
+      -id
+      "@loader_path/../lib/libssl${libsuffix}"
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl${libsuffix}"
+      COMMAND
+      install_name_tool
+      -change
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libcrypto${libsuffix}"
+      "@loader_path/../lib/libcrypto${libsuffix}"
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libssl${libsuffix}"
+      COMMAND
+      install_name_tool
+      -id
+      "@loader_path/../lib/libcrypto${libsuffix}"
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@/lib/libcrypto${libsuffix}"
+  )
+  add_dependencies(
+      "@HUNTER_EP_NAME@-fix_install_name"
+      "@HUNTER_EP_NAME@-macos_universal"
+  )
+endif()
+
+foreach(arch ${configure_architectures})
+  if("${arch}" STREQUAL "x86_64")
+    set(configure_arch "darwin64-x86_64-cc")
+  elseif("${arch}" STREQUAL "ppc64")
+    set(configure_arch "darwin64-ppc-cc")
+  elseif("${arch}" STREQUAL "i386")
+    set(configure_arch "darwin-i386-cc")
+  elseif("${arch}" STREQUAL "ppc")
+    set(configure_arch "darwin-ppc-cc")
+  else()
+    hunter_user_error("Could not find the macOS architecture ${arch}. Please update the CMAKE_OSX_ARCHITECTURES option in your toolchain.")
+  endif()
+
+  ExternalProject_Add(
+      "@HUNTER_EP_NAME@-${arch}"
+      URL
+      @HUNTER_PACKAGE_URL@
+      URL_HASH
+      SHA1=@HUNTER_PACKAGE_SHA1@
+      DOWNLOAD_DIR
+      "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+      TLS_VERIFY
+      "@HUNTER_TLS_VERIFY@"
+      SOURCE_DIR
+      "@HUNTER_PACKAGE_SOURCE_DIR@/${arch}"
+      INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+          # not used, just avoid creating Install/<name> empty directory
+      CONFIGURE_COMMAND
+      "${configure_command}"
+      "${configure_arch}"
+      "${configure_opts}"
+      "--prefix=@HUNTER_PACKAGE_INSTALL_PREFIX@"
+      BUILD_COMMAND
+      "${build_command}"
+      BUILD_IN_SOURCE
+      1
+      INSTALL_COMMAND
+      make install_sw
+          # installed libraries will be rewrited,
+          # other files are equal
+      COMMAND # Copy license files
```

Comment:
Will be called one time for each architecture, I think it make sense to call it once in `"@HUNTER_EP_NAME@-macos_universal"`. Will not make any difference but will look better I think.